### PR TITLE
ENH: modifications in prep for the documentation

### PR DESCRIPTION
**Pull request by GavinHuttley**

### CODE REVIEW

File: src/piqtree2/_app/__init__.py [hunk 1]
```diff
@@ -2,35 +2,49 @@
...
-    return build_tree(aln, model, rand_seed)
+@composable.define_app
+class piqtree_phylo:
+    @extend_docstring_from(build_tree)
+    def __init__(
+        self,
+        model: str,
+        rand_seed: int | None = None,
+    ) -> None:
+        self._model = Model(model)
```
Comment (rmcar17) [inline src/piqtree2/_app/__init__.py]: This is a breaking change (no longer allows specification of base frequencies nor rate heterogeneity)

File: src/piqtree2/_app/__init__.py [hunk 2]
```diff
@@ -2,35 +2,49 @@
...
+        return build_tree(aln, self._model, self._rand_seed)
 
 
-@define_app
+@composable.define_app
 @extend_docstring_from(fit_tree)
 def piqtree_fit(
     aln: cogent3.Alignment | cogent3.ArrayAlignment,
     tree: cogent3.PhyloNode,
     model: Model,
```
Comment (rmcar17) [inline src/piqtree2/_app/__init__.py]: Is this meant to be `str` now? (as in the new `piqtree_phylo`)

File: src/piqtree2/_app/__init__.py [hunk 3]
```diff
@@ -2,35 +2,49 @@
...
-    rand_seed: int | None = None,
+    rand_seed: int | None = None
 ) -> cogent3.PhyloNode | cogent3.app.typing.SerialisableType:
+    model = Model(model)
```
Comment (rmcar17) [inline src/piqtree2/_app/__init__.py]: This is a breaking change (no longer allows specification of base frequencies nor rate heterogeneity)

File: src/piqtree2/model/__init__.py [hunk 4]
```diff
@@ -4,7 +4,12 @@
 from ._model import Model
 from ._options import available_freq_type, available_models, available_rate_type
 from ._rate_type import DiscreteGammaModel, FreeRateModel, RateModel, RateType
-from ._substitution_model import AaModel, DnaModel, SubstitutionModel
+from ._substitution_model import (
+    AaModel,
+    DnaModel,
+    SubstitutionModel,
+    get_model,
```
Comment (rmcar17) [inline src/piqtree2/model/__init__.py]: Include in `__all__`?

**Review by rmcar17 — COMMENTED** [review]
Looks mostly good, I think we just need to think about how to support things like rate heterogeneity etc. properly with the apps

File: src/piqtree2/_app/__init__.py (hunk 1 above)
Comment (GavinHuttley) [reply src/piqtree2/_app/__init__.py]: So we need to add arguments for those components separately. For example, `rate_het: str | tuple[str]`, `state_freqs: str`. Note that `rate_het` can be itself multiple components.

Perhaps Model get's these arguments too and "compiles" a full model string for passing to `libiqtree`?

File: src/piqtree2/_app/__init__.py (hunk 2 above)
Comment (GavinHuttley) [reply src/piqtree2/_app/__init__.py]: yes, should mirror `piqtree_phylo`. I also think this should become a class. The problem with the function based apps is that the `app_help()` is not working well, does not produce a code snippet that can be copied and pasted to be a functional app/

File: src/piqtree2/_app/__init__.py (hunk 3 above)
Comment (GavinHuttley) [reply src/piqtree2/_app/__init__.py]: as above

File: src/piqtree2/model/__init__.py (hunk 4 above)
Comment (GavinHuttley) [reply src/piqtree2/model/__init__.py]: yes

File: src/piqtree2/_app/__init__.py (hunk 1 above)
Comment (rmcar17) [reply src/piqtree2/_app/__init__.py]: Something to that effect yes, need to formalise what the representation will be in that case.

Comment (GavinHuttley) [reply src/piqtree2/_app/__init__.py]: split the `rate_het` into `invariant_sites: bool=False, rate_het: str|None=None`

**Pull request merged**